zap/src/irgen/des.rs

```diff
@@ -187,6 +187,24 @@ impl Des {
 				self.push_stmt(Stmt::End);
 			}
 
+			Ty::Set(key) => {
+				self.push_assign(into.clone(), Expr::EmptyTable);
+
+				self.push_stmt(Stmt::NumFor {
+					var: "_".into(),
+					from: 1.0.into(),
+					to: self.readu16(),
+				});
+
+				let (key_name, key_expr) = self.add_occurrence("key");
+
+				self.push_ty(key, Var::Name(key_name.clone()));
```

Comment:
`key_1` isn't defined first with
`local key_1`, and so the output looks like this:
```lua
			local value
			value = {}
			for _ = 1, buffer.readu16(incoming_buff, read(2)) do
				key_1 = buffer.readu8(incoming_buff, read(1))
				value[key_1] = true
			end
```